samples/ps2dev.cmake

```diff
@@ -7,20 +24,71 @@ endif()
 SET(CMAKE_SYSTEM_NAME Generic)
 SET(CMAKE_SYSTEM_VERSION 1)
 SET(CMAKE_SYSTEM_PROCESSOR mips)
+
 SET(CMAKE_C_COMPILER mips64r5900el-ps2-elf-gcc)
 SET(CMAKE_CXX_COMPILER mips64r5900el-ps2-elf-g++)
-SET(CMAKE_C_FLAGS_INIT "-I$ENV{PS2SDK}/ee/include -I$ENV{PS2SDK}/common/include -I$ENV{PS2SDK}/ports/include -D_EE -DPS2 -D__PS2__ -O2 -G0")
-SET(CMAKE_CXX_FLAGS_INIT "-I$ENV{PS2SDK}/ee/include -I$ENV{PS2SDK}/common/include -I$ENV{PS2SDK}/ports/include -D_EE -DPS2 -D__PS2__ -O2 -G0")
-SET(CMAKE_EXE_LINKER_FLAGS_INIT "-T$ENV{PS2SDK}/ee/startup/linkfile -L$ENV{PS2SDK}/ee/lib -L$ENV{PS2DEV}/gsKit/lib -L$ENV{PS2SDK}/ports/lib -Wl,-zmax-page-size=128")
-#SET(CMAKE_SHARED_LINKER_FLAGS_INIT "...")
-#SET(CMAKE_STATIC_LINKER_FLAGS_CONFIG_INIT "...")
-#SET(CMAKE_STATIC_LINKER_FLAGS_INIT "...")
+
+SET(EE_CFLAGS "-I$ENV{PS2SDK}/ee/include -I$ENV{PS2SDK}/common/include -I$ENV{PS2SDK}/ports/include -D_EE -DPS2 -D__PS2__ -O2 -G0" CACHE STRING "EE C compiler flags" FORCE)
+SET(EE_LDFLAGS "-L$ENV{PS2SDK}/ee/lib -L$ENV{PS2DEV}/gsKit/lib -L$ENV{PS2SDK}/ports/lib -Wl,-zmax-page-size=128" CACHE STRING "EE linker flags" FORCE)
+SET(EE_ASFLAGS "-G0" CACHE STRING "EE assembler flags" FORCE)
+
+SET(EE_LINKFILE "$ENV{PS2SDK}/ee/startup/linkfile" CACHE STRING "EE linkfile" FORCE)
+
+
 SET(CMAKE_TARGET_INSTALL_PREFIX $ENV{PS2DEV}/ports)
 
 SET(CMAKE_FIND_ROOT_PATH $ENV{PS2DEV} $ENV{PS2DEV}/ee $ENV{PS2DEV}/ee/ee $ENV{PS2SDK} $ENV{PS2SDK}/ports)
 SET(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
 SET(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
 SET(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
 
+SET_PROPERTY(GLOBAL PROPERTY TARGET_SUPPORTS_SHARED_LIBS TRUE)
+
+SET(CMAKE_C_FLAGS_INIT ${EE_CFLAGS})
+SET(CMAKE_CXX_FLAGS_INIT ${EE_CFLAGS})
+SET(CMAKE_EXE_LINKER_FLAGS_INIT ${EE_LDFLAGS})
+
+SET(CMAKE_SHARED_LIBRARY_C_FLAGS)
+SET(CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS "-nostartfiles -Wl,-r -Wl,-d")
+SET(CMAKE_SHARED_LIBRARY_LINK_C_FLAGS)
```

Comment:
All the `set` here that uses standard values or don't define anything, let's remove it, the smaller we keep the file the easier to maintain